futures-util/src/future/select.rs

```diff
@@ -32,25 +32,33 @@ impl<A: Unpin, B: Unpin> Unpin for Select<A, B> {}
 ///
 /// ```
 /// # futures::executor::block_on(async {
-/// use futures::future::{self, Either};
-/// use futures::pin_mut;
-///
-/// // These two futures have different types even though their outputs have the same type
-/// let future1 = async { 1 };
-/// let future2 = async { 2 };
+/// use futures::{
+///     pin_mut,
+///     future::Either,
+///     future::self,
+/// };
+/// 
+/// // These two futures have different types even though their outputs have the same type.
+/// let future1 = async {
+///     let () = future::pending().await; // will never finish
+///     1
+/// };
+/// let future2 = async { 
+///     future::ready(2).await
+/// };
 ///
 /// // 'select' requires Future + Unpin bounds
 /// pin_mut!(future1);
 /// pin_mut!(future2);
 ///
 /// let value = match future::select(future1, future2).await {
 ///     Either::Left((value1, _)) => value1, // `value1` is resolved from `future1`
-///                                          // `_` represents `future2`
+///     // `_` represents `future2`
 ///     Either::Right((value2, _)) => value2, // `value2` is resolved from `future2`
-///                                           // `_` represents `future1`
+///     // `_` represents `future1`
```

Comment:
"`_` represents `future2`" is a description of `Either::Left((value1, _)) => value1` on the previous line, so it's probably easier to read if you format it like this:

```rust
///     // `value1` is resolved from `future1`
///     // `_` represents `future2`
///     Either::Left((value1, _)) => value1,
///     // `value2` is resolved from `future2`
///     // `_` represents `future1`
///     Either::Right((value2, _)) => value2,
```

Reply:
yeah, I haven't noticed I changed the spacing accidentally. I edited it to match the original example to:
```rust
/// let value = match future::select(future1, future2).await {
///     Either::Left((value1, _)) => value1,  // `value1` is resolved from `future1`
///                                           // `_` represents `future2`
///     Either::Right((value2, _)) => value2, // `value2` is resolved from `future2`
///                                           // `_` represents `future1`
/// };
```